pkg/testing/fixture.go

```diff
@@ -235,15 +275,20 @@ func ExtractArtifact(l Logger, artifactFile, outputDir string) error {
 	return nil
 }
 
-// Run runs the Elastic Agent.
+// Run runs the provided binary.
 //
-// If `states` are provided then the Elastic Agent runs until each state has been reached. Once reached the
+// If `states` are provided and the binary is Elastic Agent, agent runs until each state has been reached. Once reached the
 // Elastic Agent is stopped. If at any time the Elastic Agent logs an error log and the Fixture is not started
 // with `WithAllowErrors()` then `Run` will exit early and return the logged error.
 //
-// If no `states` are provided then the Elastic Agent runs until the context is cancelled.
+// If no `states` are provided then the Elastic Agent runs until the context is or the timeout specified with WithRunLength is reached.
 func (f *Fixture) Run(ctx context.Context, states ...State) error {
-	if f.installed {
+	isBeat := false
+	if f.binaryName != "elastic-agent" {
+		isBeat = true
+	}
+
+	if !isBeat && f.installed {
```

Comment:
I think it might be better to implement a `RunBeat` version of this function. Even in places where you haven't special cased on `isBeat` there is code that will never be hit when running a Beat.

For example the `stateErrCh` and `stateCh` cases in the switch statement below on line 364 don't do anything.

I think the duplication is worth not having this become confusing to read and understand. This function is written to launch a subprocess and then control what it does through the control protocol and the second part does not apply to Beats.

I also suspect that as we work more with Beats we may end up with even more specialized logic for Beats since we have no way to easily interact with it, and I don't think we should encourage that to just be bolted onto this implementation as even more special cases.